state/mysql/mysql.go

```diff
@@ -528,28 +664,29 @@ func (m *MySQL) setValue(parentCtx context.Context, querier querier, req *state.
 
 	if req.Options.Concurrency == state.FirstWrite && (req.ETag == nil || *req.ETag == "") {
 		// With first-write-wins and no etag, we can insert the row only if it doesn't exist
-		query := fmt.Sprintf(
-			`INSERT INTO %s (value, id, eTag, isbinary) VALUES (?, ?, ?, ?);`,
-			m.tableName, // m.tableName is sanitized
-		)
-		result, err = querier.ExecContext(ctx, query, enc, req.Key, eTag, isBinary)
+		query = `INSERT INTO %[1]s (value, id, eTag, isbinary, expiredate) VALUES (?, ?, ?, ?, %[2]s)`
+		params = []any{enc, req.Key, eTag, isBinary}
 	} else if req.ETag != nil && *req.ETag != "" {
 		// When an eTag is provided do an update - not insert
-		query := fmt.Sprintf(
-			`UPDATE %s SET value = ?, eTag = ?, isbinary = ? WHERE id = ? AND eTag = ?;`,
-			m.tableName, // m.tableName is sanitized
-		)
-		result, err = querier.ExecContext(ctx, query, enc, eTag, isBinary, req.Key, *req.ETag)
+		query = `UPDATE %[1]s SET value = ?, eTag = ?, isbinary = ?, expiredate = %[2]s
```

Comment:
What happens if we try to update a row that is expired but not yet garbage-collected? In that case the ETag check should fail? Or...?

Reply:
See: https://github.com/dapr/components-contrib/pull/2688